curator-client/src/main/java/org/apache/curator/ConnectionState.java

```diff
@@ -160,13 +162,33 @@ public void process(WatchedEvent event)
             }
         }
 
+        // only wait during tests
+        assert waitOnExpiredEvent(event.getState());
```

Comment:
Yes, assuming assertions are enabled in testing. I've added the delay here wait for the reconnect to happen, but since now the reset is called later in the method it might no longer be necessary.